README.md

```diff
@@ -4,227 +4,200 @@
+This component may also be referred as '3scale_backend'.
 
-### On local machine (unsupported)
+### Apisonator worker
 
-This is **unsupported** and **strongly discouraged** because it is the source of a lot
-of headaches and potential problems and introduces the need for us to document
-dependencies and config files here that are very likely to be outdated and
-incomplete.
+It is responsible for performing background tasks off-loaded from
+Apisonator Listeners (enqueued jobs).
 
-Bear in mind that we can only provide best effort support for this.
+The worker component takes care of running these enqueued jobs, mainly related
+to reporting of previous traffic.
 
-1. Install Redis v2.8.19 (or whichever version is recommended for use).
-2. Run `bundle install`.
-3. Configure Redis in `~/.3scale_backend.conf` as:
+This component needs access to a Redis database to perform the following actions:
+ * Dequeue the report jobs that have been submitted to the Redis database by the
+   Apisonator listener/s or the Apisonator failed jobs rescheduler
 
-```
-ThreeScale::Backend.configure do |config|
-  config.redis.proxy = 'localhost:6379'
-end
-```
+This component may also be referred as '3scale_backend_worker'.
 
-## Running tests
+### Apisonator failed jobs rescheduler
 
-You can either run them manually in the container-based development environment
-or have a container launched just for running the tests. For the latter you just
-need to run `make test`, and you can configure any additional environment
-variables for the test scripts like so:
+This is a simple task that acts as a cron scheduler to requeue jobs that failed
+when being processed by an Apisonator worker. The jobs are requeued into
+the Redis background jobs database.
 
-> `$ make DOCKER_OPTS="-e TEST_ALL_RUBIES=1" test`
+This component may also be referred as 'backend-cron'. 
 
-Another alternative to execute all tests from within the development environment
-where you should not need to manually start/stop the services before is to execute
-them from inside the development environment via this script:
+## Development
 
-> `$ script/test`
+See the file [DEVELOPMENT](DEVELOPMENT.md)
 
-### Running tests (advanced)
+## Deployment set up
 
-In order to run the tests manually in the container-based development, the services
-needed to run them correctly must be started before with:
+### Prerequisites
 
-> `$ script/services start`
-
-Then you can execute the following commands to execute all tests:
+* Docker (requires version 1.10.0 or later)
+* A Redis database, used to store API request statistics and services. Also
+  used to perform API requests authorizations. In Apisonator this database
+  is commonly referred as 'Redis Storage'
+* A Redis database, used to store background jobs. The Redis Resque library
+  is used for this. In Apisonator this database is commonly referred as
+  'Redis Resque', or as the 'background jobs database'
```

Comment:
maybe mention that the two databases can be combined into a single Redis instance (by using the database id specification, ie. `redis://host:port/0`, `redis://host:port/1`, etc